### PR TITLE
Fix incorrect workaround for pure Y (X) rotations in XYZ (YXZ) conven…

### LINKED ISSUE #10352: Applying scale to a spatial turns it upside down
**Operating system or device - Godot version:**
Linux, 53c23b02226968d27e6caadcb801343697ac4fe9

**Issue description:**
When we apply scale to the model that is looking in z positive direction, in a result it will be turned upside down: 
upside_down

I did bisection and I found that 53c23b02226968d27e6caadcb801343697ac4fe9 introduced this behavior 

**Steps to reproduce:**
1. Download reproduction project.
2. Open `res://Issues/YFlipIssue/YFlipIssue.tscn`
3. Run it.

Alternatively create new spatial and add this code as it's script:
```
func _ready():
	look_at(global_transform.origin + Vector3(0,0,1), Vector3(0,1,0));

func _process(delta):
	var lscale = get_scale();
	set_scale(lscale)
```

**Link to minimal example project:**
[upsidedown_issue.zip](https://github.com/godotengine/godot/files/1225347/upsidedown_issue.zip)

### PR DESCRIPTION
**Pull request by ghost**
…tion.

Fixes #10352.

### CODE REVIEW

File: core/math/matrix3.cpp [hunk 1]
```diff
@@ -436,15 +432,10 @@ Vector3 Basis::get_euler_yxz() const {
 
 	if (m12 < 1) {
 		if (m12 > -1) {
-			if (elements[1][0] == 0 && elements[0][1] == 0 && elements[2][2] < 0) { // use pure x rotation
-				real_t x = asin(-m12);
+			if (elements[1][0] == 0 && elements[0][1] == 0 && elements[0][2] == 0 && elements[2][0] == 0 && elements[0][0] == 1) { // use pure x rotation
+				real_t x = atan2(-m12, elements[1][1]);
```
Comment (akien-mga) [inline core/math/matrix3.cpp]: `x` is unused since you removed the chunk below, I guess it should be `euler.x`?

Comment (tagcup) [inline core/math/matrix3.cpp]: Indeed, thanks!

**Comment by tagcup** [comment]
This of course doesn't fix the problem regarding the non-uniqueness of polar decomposition, which makes it impossible to change scaling without changing rotation part, vice versa.

The behavior of Spatial::set_scale was changed in #7438, where I introduced polar decomposition in Basis::get_scale and Basis::get_rotation (because earlier behavior wasn't mathematically correct in general). So Spatial::set_scale in 2.x and 3.x does different things.

I'll remove (or at least unexport to GDScript) Basis::set_scale and Basis::set_rotation, which I introduced in #8277 in a later PR, and their get_ counterparts need some thinking as well.

**Comment by reduz** [comment]
oh, nice, fixes makes a lot more sense than what I did

**Comment by tagcup** [comment]
Clarification for others: this simply checks whether we have a [simple rotation around the Y (X) axis](https://en.wikipedia.org/wiki/Rotation_matrix#Basic_rotations) (there should be 4 zeroes and a single 1), and proceeds to extract the rotation angle.

**Pull request merged**